src/components/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { userActions } from '../../actions';
+
+class RegisterPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {
+        name: '',
+        email: '',
+        password: '',
+      },
+    //   submitted: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    const { user } = this.state;
+    this.setState({
+      user: {
+        ...user,
+        [name]: value,
+      },
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    // this.setState({ submitted: true });
+    const { user } = this.state;
+    const { register } = this.props;
+    const { email, password, name } = user;
+    if (email && password && name) {
+      register(email, password);
+    }
+  }
+
+  render() {
+    const { registering } = this.state;
+    const { email, name, password } = this.state;
+    return (
+      <div>
+        <h2>
+            Register
+        </h2>
+        <form name="form" onSubmit={this.handleSubmit}>
+          <label htmlFor="name">
+            Name
+            <input type="text" name="name" value={name} onChange={this.handleChange} />
+          </label>
+          <label htmlFor="email">
+            Email
+            <input type="text" name="email" value={email} onChange={this.handleChange} />
+          </label>
+          <label htmlFor="password">
+            Password
+            <input type="password" name="password" value={password} onChange={this.handleChange} />
+          </label>
+          <button type="submit">
+            Register
+          </button>
+          {registering && (
+          <h2>
+            Loading
+          </h2>
+          ) }
+          <NavLink to="/login">
+           Cancel
+          </NavLink>
+        </form>
+      </div>
+    );
+  }
+}
+// submitted is used here to do theh checks if the input fields are empty or not.
+const mapStateToProps = (state) => {
+  const { registering } = state;
+  return {
+    registering,
```

Comment:
this is going to add a `registering` prop but in the component, it looks like you are expecting `registering` to be part of the components state. redux "state" !== component "state". they are two totally different things, just happen to both use the term "state"

Reply:
Got it, I was a little confused there. Thanks for letting me know. Fixed that.